### PR TITLE
[SPARK-5773][Mllib] Further optimize sparse syr

### PR DESCRIPTION
**Pull request by viirya**
The current sparse syr in BLAS can be further optimized.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala [hunk 1]
```diff
@@ -276,11 +276,13 @@ private[spark] object BLAS extends Serializable with Logging {
 
     var i = 0
     while (i < nnz) {
-      val multiplier = alpha * xValues(i)
-      val offset = xIndices(i) * mA
-      var j = 0
+      var j = i
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala]: If you start at `i+1`, you don't even need to check if `i != j`. Is this notably faster at reasonable scale?

Comment (viirya) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala]: Ah, yes. I did not notice that. Thanks.

**Comment by srowen** [comment]
I think it's worth a little benchmarking, since it's easy, to confirm whether it's faster. It does change the memory access pattern.

**Comment by viirya** [comment]
Okay. I would do it later. Roughly considering it, the half computation can be saved.

**Comment by viirya** [comment]
Performance is not improved. Close this pr.

**Pull request closed without merging**